scope.go

```diff
@@ -72,10 +76,14 @@ type scope struct {
 	tm sync.RWMutex
 	hm sync.RWMutex
 
-	counters   map[string]*counter
-	gauges     map[string]*gauge
-	timers     map[string]*timer
-	histograms map[string]*histogram
+	counters        map[string]*counter
+	countersSlice   []*counter
```

Comment:
nit: Can we find a better name than `fooSlice`?

Reply:
thought about a few different choices:
- `fooList`, `fooCollection`, etc: all seem worse than `fooSlice`.
- `fooIteration` or something along those lines to indicate it's for iteration but that seems even more artificial

need this for something so going to land as is. if you have a better suggestion, i'll open a followup.